contentgrid-spring-audit-logging/src/main/java/com/contentgrid/spring/audit/extractor/EntityEventExtractor.java

```diff
@@ -0,0 +1,39 @@
+package com.contentgrid.spring.audit.extractor;
+
+import com.contentgrid.spring.audit.event.AbstractEntityAuditEvent.AbstractEntityAuditEventBuilder;
+import com.contentgrid.spring.audit.event.AbstractAuditEvent.AbstractAuditEventBuilder;
+import java.util.Map;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.rest.core.mapping.RepositoryResourceMappings;
+import org.springframework.data.rest.core.mapping.ResourceMetadata;
+import org.springframework.http.server.observation.ServerRequestObservationContext;
+import org.springframework.web.servlet.HandlerMapping;
+
+@RequiredArgsConstructor
+public class EntityEventExtractor implements AuditEventExtractor{
+
+    private final RepositoryResourceMappings resourceMappings;
+
+    @Override
+    public Optional<AbstractAuditEventBuilder<?, ?>> createEventBuilder(ServerRequestObservationContext context) {
+        return Optional.empty();
+    }
+
+    @Override
+    public AbstractAuditEventBuilder<?, ?> enhance(ServerRequestObservationContext context,
+            AbstractAuditEventBuilder<?, ?> eventBuilder) {
+        if(eventBuilder instanceof AbstractEntityAuditEventBuilder<?, ?> entityAuditEventBuilder) {
+            Map<String, String> templateVariables = (Map<String, String>) context.getCarrier()
+                    .getAttribute(HandlerMapping.URI_TEMPLATE_VARIABLES_ATTRIBUTE);
+
+            for (ResourceMetadata resourceMapping : resourceMappings) {
+                if (resourceMapping.isExported() && resourceMapping.getPath().matches(templateVariables.get("repository"))) {
```

Comment:
how do you know `templateVariables` is not going to be `null` ?

Reply:
It is set by `RequestMappingInfoHandlerMapping`, which is a base class of the HandlerMapping classes used by spring-content, spring-data-rest, and by any custom controllers that use annotations.
It would only not be present when there is a custom `HandlerMapping` implementation; in which case the auditing won't match at all.